ls1intum/Artemis
# `Development`: Modernize appearance of plagiarism similarity distribution and tutor effort statistics

### Checklist
#### General


#### Client

### Motivation and Context


This PR changes the implementation of the plagiarism similarity distribution chart and the tutor effort statistics chart from `ng2-charts`/ `chart.js to ngx-charts`.

### Description

The html templates as well as the underlying logic has been refactored in order to fit the requests of `ngx-charts`.
### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Text exercise that already has been assessed by tutors and with at least one plagiarism case.

**Important**: As always, take a look at the old implementation first. Then compare it to the new one.

1. Log in to Artemis
2. Navigate to Course Administration
3. Enter the exercise overview
4. For the text exercise, click on the exercise title
5. Click on "Check plagiarism"
6. If the plagiarism is not detected yet, run the plagiarism detection and confirm the case(s)
7. Click on "Run details" right above the first case
8. Inspect the displayed chart
9. Return to the exercise detail page and click "Tutor Effort Statistics"
10. Inspect the chart

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


Old implementation of Similarity Distribution:
![image](https://user-images.githubusercontent.com/80622272/146539181-f7bd2f3c-6af8-4ff2-8c9f-c8b5f137f025.png)

New implementation of Similarity Distribution:
![image](https://user-images.githubusercontent.com/80622272/146539124-54554f9a-263d-4f04-9d45-04a4f2601042.png)


Old implementation of Tutor Effort Statistics:
![image](https://user-images.githubusercontent.com/80622272/146175700-4e0fc098-9c0d-4a38-a5dd-7f7768ff7d17.png)

New implementation of Tutor Effort Statistics:
![image](https://user-images.githubusercontent.com/80622272/146539094-345c00f4-f593-46e1-8781-353a8996d71a.png)
